internal/pkg/deploy/cloudformation/stack/static_site.go

```diff
@@ -100,9 +100,12 @@ func (s *StaticSite) Template() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	bucket, path, err := s3.ParseURL(s.assetMappingURL)
-	if err != nil {
-		return "", err
+	var bucket, path string
+	if s.assetMappingURL != "" {
```

Comment:
When is this empty? when there is no asset to upload?

Reply:
> when there is no asset to upload?

When we don't perform assets uploading for example `svc package` without `--upload-assets`

Reply:
qq - is `bucket` and `path` computable just from the hash of all assets, even if they are not actually uploaded?

Reply:
Yeah that's true but it won't be useful anyway to include in the template I think. The same logic could be applied to the other uploadable like custom resources, addons, and env files.

Reply:
The other uploadable's URLs are precomputed and filled into the template, even without `--upload-assets` though! 

I am okay with not writing the URLs because I don't have a good argument against it right now. I feel like there is a difference between "providing an invalid template" vs. "providing a valid template that may not work" - not writing the precomputed URLs is the former, writing them without necessarily uploading them is the later. Though I don't see how realistically it's a huge difference in terms of usability.

Reply:
> The other uploadable's URLs are precomputed and filled into the template, even without --upload-assets though!

I don't think I follow. I just tested it out and I got

```
  TriggerStateMachineFunction:
    Metadata:
      aws:copilot:description: A lambda that triggers the file-copying state machine
    Type: AWS::Lambda::Function
    Properties:
      Handler: index.handler
      Role: !GetAtt TriggerStateMachineFunctionRole.Arn
      Runtime: nodejs16.x
      Timeout: 900
      MemorySize: 512
      Code:
        S3Bucket: 
        S3Key:

Reply:
https://github.com/aws/copilot-cli/pull/4509/files I'm thinking about this PR

Reply:
Gotcha I see. We should probably add that for static site too. Maybe for custom resources it's ok because they should be something copilot handles under the hood where as static assets are more similar to env files/addons template URL/ECR image?